packages/vite/src/node/server/middlewares/transform.ts

```diff
@@ -213,6 +213,7 @@ export function transformMiddleware(
         '',
       )
       if (
+        !url.startsWith('/@id/\0') &&
```

Comment:
At this point, we have replaced the `NULL_BYTE_PLACEHOLDER` but didn't yet unwrapped the id. We could actually remove the whole check now that we are doing it before load but leaving that to be done in a separate PR.